### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by mshd**

### CODE REVIEW

File: apps/admin/types/roles.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { prismaClient } from "db";
+import { Permission } from "ra-data-simple-prisma";
+import { Role } from "@prisma/client";
+
+export type Permissions = Permission<keyof typeof prismaClient>[];
+
+export type PermissionsConfig = {
+  [role in Role]: Permission<keyof typeof prismaClient>[];
```
Comment (ogroppo) [inline apps/admin/types/roles.ts]:
```suggestion
  [role in Role]: Permissions;
```

File: apps/admin/middlewares/extract.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { getSession } from "next-auth/react";
+import {
+  fetchActionToAction,
+  ReactAdminFetchActions,
+  Permissions,
+} from "ra-data-simple-prisma";
+import { authProvider } from "../providers/authProvider";
+
+export const extractAccessObjFromReq = async (req: NextApiRequest) => {
+  const session = await getSession({ req });
```
Comment (ogroppo) [inline apps/admin/middlewares/extract.ts]: does it not work with getServerSession?

File: apps/admin/middlewares/extract.ts
Comment (ogroppo) [inline apps/admin/middlewares/extract.ts:1]: this is not actually used as a middleware

Comment (mshd) [reply apps/admin/middlewares/extract.ts:1]: true

File: apps/admin/middlewares/extract.ts (hunk 2 above)
Comment (mshd) [reply apps/admin/middlewares/extract.ts]: not tested yet, but this is the old code, which should work

Comment (mshd) [reply apps/admin/middlewares/extract.ts]: fixed, changed to getServerSession

File: apps/admin/.babelrc
Comment (mshd) [inline apps/admin/.babelrc:1]: not needed right?

File: apps/admin/providers/accessCheck.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  fetchActionToAction,
+  ReactAdminFetchActions,
+  Permissions,
+  canAccess,
+} from "ra-data-simple-prisma";
+import { authProvider } from "./authProvider";
+import { authOptions } from "../pages/api/auth/[...nextauth]";
+import { getServerSession } from "next-auth";
+
+export const extractAccessObjFromReq = async (
```
Comment (mshd) [inline apps/admin/providers/accessCheck.ts:12]: this is the only file that I couldn't abstract to be used in the package

File: apps/admin/.babelrc
Comment (ogroppo) [reply apps/admin/.babelrc:1]: It's actually needed for jest, but there are not tests ATM

File: apps/admin/custom-pages/Dashboard.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+export default function Dashboard() {
+  const { data: session, status } = useSession();
+
+  return (
+    <Box>
+      <Typography variant="h2">Dashboard</Typography>
+      <Typography my={3}>
+        Welcome {session?.user?.name}.
+        <br /> You are connected to the database:
+        <br /> Your role is: {session?.role}
```
Comment (ogroppo) [inline apps/admin/custom-pages/Dashboard.tsx:14]: 👍

File: apps/admin/custom-pages/LoginPage.tsx [hunk 5]
```diff
@@ -40,10 +44,17 @@ const LoginPage = () => {
...
             Sign in
           </Typography>
-          <Box component="form" noValidate sx={{ mt: 1 }}>
+          <Box
+            component="form"
+            noValidate
+            sx={{ mt: 1 }}
+          >
             <Button
               // type="submit"
```
Comment (ogroppo) [inline apps/admin/custom-pages/LoginPage.tsx]: this can be removed

File: apps/admin/custom-pages/LoginPage.tsx [hunk 6]
```diff
@@ -54,6 +65,7 @@ const LoginPage = () => {
               Sign In with Google
             </Button>
           </Box>
+          <Link href="/api/auth/signin">Login by password</Link>
```
Comment (ogroppo) [inline apps/admin/custom-pages/LoginPage.tsx]: is this the next-auth form?

File: apps/admin/pages/api/[resource].ts [hunk 7]
```diff
@@ -1,5 +1,5 @@
 import type { NextApiRequest, NextApiResponse } from "next";
-import { defaultHandler } from "ra-data-simple-prisma";
+import { defaultHandler, canAccess } from "ra-data-simple-prisma";
```
Comment (ogroppo) [inline apps/admin/pages/api/[resource].ts]: unused import

File: apps/admin/providers/accessCheck.ts (hunk 3 above)
Comment (ogroppo) [reply apps/admin/providers/accessCheck.ts:12]: right, it's too dependent on nextjs

**Review by ogroppo — APPROVED** [review]

File: apps/admin/custom-pages/LoginPage.tsx (hunk 6 above)
Comment (mshd) [reply apps/admin/custom-pages/LoginPage.tsx]: yes but can be removed

File: packages/ra-data-simple-prisma/package.json [hunk 8]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "ra-data-simple-prisma",
-  "version": "2.2.1",
-  "description": "No brainer react-admin dataprovider for prisma, now also supporting audit logs",
+  "version": "2.3.0",
+  "description": "Simple react-admin dataprovider for prisma, supporting audit logs and roles",
```
Comment (ogroppo) [inline packages/ra-data-simple-prisma/package.json:4]: nice

**Pull request merged**